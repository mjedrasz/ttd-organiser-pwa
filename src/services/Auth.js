import Auth from '@aws-amplify/auth';
import JS from '@aws-amplify/core/lib/JS';

import { SIGNED_IN, VERIFY_CONTACT } from '../constants/authStates';

const GOOGLE_PROVIDER = 'Google';
const FACEBOOK_PROVIDER = 'Facebook';

export class AuthClass {

    confirmSignUp = async (username, code) => {
        await Auth.confirmSignUp(username, code);
    };

    resendCodeSignUp = async (username) => {
        await Auth.resendSignUp(username);
    };

    forgotPassword = async (username) => {
        await Auth.forgotPassword(username)
    };

    resetPassword = async (username, code, password) => {
        await Auth.forgotPasswordSubmit(username, code, password);
    };

    resendCode = async (username) => {
        await Auth.forgotPassword(username);
    };

    signIn = async (email, password) => {
        const user = await Auth.signIn(email, password);
        return await this.checkContact(user);
    };

    checkContact = async (user) => {
        const data = await Auth.verifiedContact(user);
        if (!JS.isEmpty(data.verified)) {
            return {
                state: SIGNED_IN,
                data: user
            };
        } else {
            return {
                state: VERIFY_CONTACT,
                data: { ...user, ...data }
            };
        }
    };

    socialSignIn = (provider) => {
        const config = Auth.configure();

        const {
            userPoolWebClientId,
            oauth: {
                scope,
                domain,
                redirectSignIn,
                responseType
            }
        } = config;
        const url = `https://${domain}/oauth2/authorize?identity_provider=${provider}&redirect_uri=${redirectSignIn}&response_type=${responseType}&client_id=${userPoolWebClientId}&scope=${scope.join('+')}`;
        window.location.assign(url);
    }

    googleSignIn = () => {
        this.socialSignIn(GOOGLE_PROVIDER);
    }

    facebookSignIn = () => {
        this.socialSignIn(FACEBOOK_PROVIDER);
    }

    signUp = async (username, password, attributes = {}) =>
        await Auth.signUp({
            username,
            password,
            attributes
        });


    signOut = async () => await Auth.signOut({ global: true });
}

const AuthService = new AuthClass();

export default AuthService;