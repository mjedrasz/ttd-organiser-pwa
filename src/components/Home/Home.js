import React from 'react';
import clsx from 'clsx';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ExitToApp from '@material-ui/icons/ExitToApp';
import { mainListItems } from '../MenuItems/MenuItems';
import useStyles from './styles';
import Cards from '../Drafts/Drafts';
import { HashRouter as Router, Route } from 'react-router-dom';
import Auth from '../../services/Auth';
import I18n from "@aws-amplify/core/lib/I18n";


export default () => {

  const classes = useStyles();

  const [open, setOpen] = React.useState(true);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleSignOut = async () => {
    await Auth.signOut();
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <Router>
      <div className={classes.root}>
        <CssBaseline />
        <AppBar position="fixed" className={clsx(classes.appBar, open && classes.appBarShift)}>
          <Toolbar className={classes.toolbar}>
            <IconButton
              edge="start"
              color="inherit"
              aria-label="open drawer"
              onClick={handleDrawerOpen}
              className={clsx(classes.menuButton, open && classes.menuButtonHidden)}
            >
              <MenuIcon />
            </IconButton>
            <Typography component="h1" variant="h6" color="inherit" noWrap className={classes.title}>
              {I18n.get("Dashboard")}
          </Typography>
            <IconButton color="inherit" onClick={handleSignOut} aria-label="sign out">
                <ExitToApp />
            </IconButton>
          </Toolbar>
        </AppBar>


        <Route>
          <Drawer
            variant="permanent"
            classes={{
              paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
            }}
            open={open}
          >
            <div className={classes.toolbarIcon}>
              <IconButton onClick={handleDrawerClose}>
                <ChevronLeftIcon />
              </IconButton>
            </div>
            <Divider />
            <List>{mainListItems}</List>
          </Drawer>
          <Route exact path='/'>
              <div className={classes.appBarSpacer} />
                  <Cards />
            </Route>
        </Route>

      </div>
    </Router>
  );
}