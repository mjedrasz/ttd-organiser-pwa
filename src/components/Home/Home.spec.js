import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';
import Auth from '../../services/Auth';
import { render } from '@testing-library/react';

jest.mock('../Drafts/Drafts', () =>
  jest.fn(() => (
    <div>
      Things To Do
    </div>
  )));

jest.mock('../../services/Auth', () => ({
  signOut: jest.fn()
}));

describe('Home', () => {

  test('snapshot renders correctly', () => {
    const component = renderer.create(<Home />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call sign out function on sign out icon click', async () => {

    const mockSignOut = jest.fn();
    Auth.signOut.mockImplementation(mockSignOut);

    const { getByLabelText } = render(<Home />);

    const signOut = getByLabelText('sign out');

    signOut.click();

    expect(mockSignOut).toBeCalled();

  });

});