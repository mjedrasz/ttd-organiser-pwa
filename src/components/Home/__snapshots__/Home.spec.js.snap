// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-4 makeStyles-appBarShift-5 MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 makeStyles-menuButtonHidden-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h1
        className="MuiTypography-root makeStyles-title-8 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        Dashboard
      </h1>
      <button
        aria-label="sign out"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-drawerPaper-9 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
    >
      <div
        className="makeStyles-toolbarIcon-3"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div>
          <a
            href="#/"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2v7.22z"
                  />
                  <path
                    d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3V2zM7 9h8v2H7zM7 12v2h8v-2h-3zM7 15h8v2H7z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                >
                  Drafts
                </span>
              </div>
            </div>
          </a>
          <a
            href="#/"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                  <path
                    d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                >
                  Pending
                </span>
              </div>
            </div>
          </a>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Published
              </span>
            </div>
          </div>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 3H6c-.83 0-1.54.5-1.84 1.22l-3.02 7.05c-.09.23-.14.47-.14.73v2c0 1.1.9 2 2 2h6.31l-.95 4.57-.03.32c0 .41.17.79.44 1.06L9.83 23l6.59-6.59c.36-.36.58-.86.58-1.41V5c0-1.1-.9-2-2-2zm4 0v12h4V3h-4z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                Rejected
              </span>
            </div>
          </div>
        </div>
      </ul>
    </div>
  </div>
  <div
    className="makeStyles-appBarSpacer-12"
  />
  <div>
    Things To Do
  </div>
</div>
`;
